### PR TITLE
Calculate Physical Input Byte Size for queries in MongoDB Connector

### LINKED ISSUE #17713: Calculate Physical Input size for query stats in Mongo DB connector
Mongo DB connector does not calculate physical input size and always returns 0.

https://github.com/trinodb/trino/blob/a0eb0a107abd432458f3954619610bf63c92c10f/plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java#L113-L117

This Physical Input size will be helpful for adding a test case for dereference pushdown (https://github.com/trinodb/trino/pull/17710) comparing physical input size with and without enabling dereference pushdown.

### PR DESCRIPTION
**Pull request by academy-codex**
## Description
Fixes #17713
Currently the MongoDB Connector returns 0 for the getCompletedBytes() in the PageSource implementation.
The suggested approach is to capture the total bytes by take the JSON String representation of the Mongo Document and converting it into a bytes array and simply taking it's length.

## Additional context and related issues
N/A

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java [hunk 1]
```diff
@@ -113,7 +115,7 @@ public MongoPageSource(
     @Override
     public long getCompletedBytes()
     {
-        return 0;
+        return totalBytes;
```
Comment (ebyhr) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java:118]: Please check CI failure.

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java [hunk 2]
```diff
@@ -144,6 +146,7 @@ public Page getNextPage()
                 break;
             }
             currentDoc = cursor.next();
+            totalBytes += currentDoc.toJson().getBytes(StandardCharsets.UTF_8).length;
```
Comment (ebyhr) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java:149]: Static import `UTF_8`.

Comment (ebyhr) [reply plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java:149]: Calling `toJson` method increases memory usage?

Comment (Praveen2112) [reply plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java:149]: Does this JSON represents the actual bytes read from the MongoDB or a different interpretation of the data read from MongoDB ?

Comment (academy-codex) [reply plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java:149]: > Does calling `toJson` method increase CPU & memory usage?

Overall, while there is some impact on CPU and memory usage when invoking the toJSON() method, it is generally considered to be minor and well-optimized. The impact becomes more noticeable for larger or more complex documents, but it is still within acceptable limits for most use cases. In most cases, it should be unnoticeable.

1. CPU Usage: The CPU impact of the toJSON() method is typically minimal for small and simple documents. While this process incurs some CPU overhead, it is generally efficient and designed to be performant. 

2. Memory Usage: The memory impact of the toJSON() method is also generally minor. The memory required to store the resulting JSON string is temporary and limited to the scope of the method invocation.

Comment (Praveen2112) [reply plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java]: Does `toJson` method captures the approximate size of the input data or does it include the Json parts like `{` as well ? This `toJson` will be invoked for each row for a pretty big dataset won't it be an overhead ?

**Comment by hashhar** [comment]
This adds a lot of overhead. FWIW in unrelated place we've seen serializing JSON as taking ~30% of CPU time on a node.

It should be solved in alternate way if possible.

**Pull request closed without merging**